### DISCUSSION TITLE
driver.quit() problem

### DISCUSSION THREAD

**Discussion by tromotixc** [ultrafunkamsterdam/undetected-chromedriver#1665] [Q&A]
Hello,

I use this way:
` chrome_options = uc.ChromeOptions()
  chrome_options.add_argument("--disable-extensions")
  chrome_options.add_argument("--disable-popup-blocking")
  chrome_options.add_experimental_option("prefs", {
      "credentials_enable_service": False,
      "profile.password_manager_enabled": False
  })

  driver = uc.Chrome(options=chrome_options)`

but when I would like to driver.quit() it closes the window but in the task manager I still see the Crome running my CPU

is there any solution for it?

thank you in advance!

**Comment by zmttmz** [comment]
Hi,
Yup. Noticed this long time ago, i am running it on CentOS 7. Don't know if it is related to operating system but for sure all opened UC when using driver.quit() they still leave a zombie process. In my research i just found something that it might be related to the CentOS, but if you are using other system then other is the problem. Thanks!

**Reply by Sleeeepy7** [reply]
Hi,
On a linux ubuntu 20.0.3 i have the same problem, every execution leave a zombie process, do you have any solutions or tips :?

**Reply by zmttmz** [reply]
Hi,
Unfortunately couldn't find any working solution for this matter

**Reply by MrPakoras** [reply]
Yep, I have this issue too. I didnt realise until I tried running a game and the FPS was horrible. Ran taskkill /F -im "chrome.exe" to find that there were like 50-100 chrome processes running lol (even with the browser closed).

**Reply by WimP99** [reply]
similar problem on win10

**Reply by zmttmz** [reply]
Hello,
After a little more digging and errors recording i have found the following. 

Something is not working properly with the undetected_webdriver.

Tested first on selenium webdriver with chrome and i catch the same error but the behavior of the webdriver and UC are different. 

In the selenium webdriver when you parse many links, occasionally there is an error with the webdriver that it couldn't be started and the browser couldn't be started.  So i do the return and put it for later try, everything good no remaining opened browsers. Since it occurs very rarely is not that big of a deal, though i have no idea what is causing the error with the exit code 1 for the webdriver. 

The things are changing when using UC, the same error occurs on UC saying that couldn't start the undetected_chromedriver and also the browser couldn't be started but the behavior is different, although is saying the browser couldn't be started it is in fact starting and remains started as it is not used with the UC process so after return it stays started with no way to close it (or at least i have no idea how). When the UC is seeing the browser successfully started then it can be closed with browser.quit() (working). But in this case when the UC gives the error that couldn't start it then the browser is None and the quit() obviously not working. 

If anyone has some thoughts about it please say :)

**Reply by KhaldiDataPanda** [reply]
same issue here on win10, i both used .close() before .quit() and alone .quit() , it's always an OSError, that doesn't occur with regular slenium

**Reply by PostMalone** [reply]
Same issue, started happening to me 2-3 days ago.

Even though I'm using driver.quit() - the Chrome process in still there in Task Manager, clogging up the CPU usage.

**Comment by josetilin37** [comment]
> Hello,
> 
> I use this way: ` chrome_options = uc.ChromeOptions() chrome_options.add_argument("--disable-extensions") chrome_options.add_argument("--disable-popup-blocking") chrome_options.add_experimental_option("prefs", { "credentials_enable_service": False, "profile.password_manager_enabled": False })
> 
> driver = uc.Chrome(options=chrome_options)`
> 
> but when I would like to driver.quit() it closes the window but in the task manager I still see the Crome running my CPU
> 
> is there any solution for it?
> 
> thank you in advance!

hello, in windows 10 I had the same problem about the phantom processes and I managed to solve it by using pyautogui to close the browser just before applying the driver.quit(), this way the browser is completely closed just before the driver.quit() and leaves no trace in the task manager, this solution was effective for me. I hope this helps you.

```
pyautogui.moveTo(x, y, duration=0.35) #coords for closing the browser
pyautogui.click()
driver.quit()
```

You can adjust the duration of this action to make it more instantaneous
